util/reflect.go

```diff
@@ -545,6 +548,108 @@ func forEachFieldInternal(ni *NodeInfo, in, out interface{}, iterFunction FieldI
 	return errs
 }
 
+// ForEachDataField iterates the value supplied and iterates iterFunction for
+// each data tree node found in the function. No schema information is required
+// to perform the iteration.
+func ForEachDataField(value, in, out interface{}, iterFunction FieldIteratorFunc) Errors {
+	if IsValueNil(value) {
```

Comment:
(optional) Is this check redundant? The forEachDataFieldInternal() func seems to start with the same check.

Reply:
No - the check here and the check in the `forEachDataFieldInternal` are actually for different things -- the check in the `forEachDataFieldInternal` checks whether the supplied `NodeInfo` is nil - rather than whether the input value is `nil`. I think it's worth having both since we'll get either a panic (`reflect.ValueOf` on a nil value causes a panic) or a NPE without both these checks.